packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2926,6 +2926,28 @@ export class ContainerRuntime
 					error,
 				};
 			}
+
+			// If validateSummaryBeforeUpload is true, validate that the summary generated by the summarizer nodes is
```

Comment:
This is just a refactor. Moved this validation up the stack and removed properties in returned object that are not needed for "base" stage result.

Reply:
Wouldn't this actually be "generate" stage since the summary was actually generated?

Reply:
I consider validation to be part of summary generation. Yes, technically the summary was generated but if validation step fails, the summary was incorrect and cannot be trusted. I consider that to be a failed summary generation.

Also, FYI stage here means the stage upto which summarization succeeds.

Reply:
Ok, sounds good to me :)